nbi/buildspec.yml

```diff
@@ -20,14 +20,14 @@ phases:
       - echo "Setting up conda"
       - export PATH=$LCC_CONDA_INSTALL_DIR/bin:$PATH
       - for CHANNEL in conda-forge cf-staging; do conda config --system --append channels $CHANNEL; done
-      - conda install --yes conda-pack
   build:
     commands:
       - BRAKET_ENV=Braket
       - mkdir -p envs
       - conda config --set path_conflict warn
-      - conda env create --name $BRAKET_ENV -f environment.yml
-      - conda pack --name $BRAKET_ENV --output envs/Braket.tgz
+      - conda env create --name $BRAKET_ENV -f environment.yml --no-default-packages
+      - conda list
+      - conda pack --name $BRAKET_ENV --output envs/Braket.tgz --ignore-missing-files
```

Comment:
What are these changes for?

Reply:
I believe adding `--no-default-packages` doesn't install the conda default packages while creating an environment? 
For the --no-default-package options, we would need to determine if we are anyways installing those dependencies later. From what I know conda installs dependencies like python, pip, setup-tools, wheel, numpy, scipy, matplotlib and pandas, and few others which are generally required by other packages. You can get the list of packages installed by conda, when you do `conda list` with default created environment. 

I don't think we should be using the `--ignore-missing-files` option as it can cause missing dependencies which we would have to keep testing the NBIs or would manually add them every time? Thoughts?

Reply:
I am still hesitant about using `--ignore-missing-files` as it can cause problems. If we accidentally misspell the package name too it ignore that considering that package is not available. If there are any transitive dependency issues, we won't be able to catch that as well and might lead to incomplete environment creation.